File: Serene/Serene.Script/Northwind/Customer/CustomerOrdersGrid.cs

namespace Serene.Northwind
{
    using System.Collections.Generic;
    using System.Linq;
    using jQueryApi;
    using Serenity;

    [DialogType(typeof(CustomerOrderDialog))]
    public class CustomerOrdersGrid : OrderGrid
    {
        public CustomerOrdersGrid(jQueryObject container)
            : base(container)
        {
        }

        protected override List<SlickColumn> GetColumns()
        {
            return base.GetColumns().Where(x => x.Field != OrderRow.Fields.CustomerCompanyName).ToList();
        }

        protected override void InitEntityDialog(string itemType, Widget dialog)
        {
            base.InitEntityDialog(itemType, dialog);
            Serenity.SubDialogHelper.Cascade((OrderDialog)dialog, this.Element.Closest(".ui-dialog"));
        }

        protected override void AddButtonClick()
        {
            EditItem(new OrderRow
            {
                CustomerID = CustomerID
            });
        }

        protected override string GetInitialTitle()
        {
            return null;
        }

        protected override void CreateToolbarExtensions()
        {
            base.CreateToolbarExtensions();

            CustomerFilter.Element.Closest(".quick-filter-item").Remove();
        }

        protected override bool GetGridCanLoad()
        {
            return base.GetGridCanLoad() && !string.IsNullOrEmpty(customerID);
        }

        private string customerID;

        public string CustomerID
        {
            get { return customerID; }
            set
            {
                if (customerID != value)
                {
                    customerID = value;
                    SetEquality(OrderRow.Fields.CustomerID, CustomerID);
                    Refresh();
                }
            }
        }
    }
}

File: Serene/Serene.Web/Modules/Northwind/Customer/CustomerOrdersGrid.ts
/// <reference path="../Order/OrderGrid.ts" />

namespace Serene.Northwind {

    @Serenity.Decorators.registerClass()
    export class CustomerOrdersGrid extends OrderGrid {
        protected getDialogType() { return CustomerOrderDialog; }

        constructor(container: JQuery) {
            super(container);
        }

        protected getColumns() {
            let fld = OrderRow.Fields;
            return super.getColumns().filter(x => x.field !== fld.CustomerCompanyName);
        }

        protected initEntityDialog(itemType, dialog) {
            super.initEntityDialog(itemType, dialog);
            Serenity.SubDialogHelper.cascade(dialog, this.element.closest('.ui-dialog'));
        }

        protected addButtonClick() {
            this.editItem({ CustomerID: this.customerID });
        }

        protected getInitialTitle() {
            return null;
        }

        protected createQuickFilters() {
            super.createQuickFilters();
            this.customerFilter.element.closest('.quick-filter-item').remove();
        }

        protected getGridCanLoad() {
            return super.getGridCanLoad() && !!this.customerID;
        }

        private _customerID: string;

        get customerID() {
            return this._customerID;
        }

        set customerID(value: string) {
            if (this.customerID !== value) {
                this.customerID = value;
                this.setEquality('CustomerID', value);
                this.refresh();
            }
        }
    }
}